test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -137,6 +138,13 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, nil
 	}
 
+	// Check if control plane is initialized, otherwise return and wait for the control plane to be available.
+	if !cluster.Status.ControlPlaneInitialized {
+		log.Info("Waiting for DockerCluster control plane to become available")
+		conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, bootstrapv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
I don't think we should use costs from bootstrapv1 in CAPD.
Given that WaitingForControlPlaneAvailableReason is going to be used in several providers, my suggestion is to
- deprecate bootstrapv1.WaitingForControlPlaneAvailableReason
- create clusterv1.WaitingForControlPlaneAvailableReason (in cluster-api/api/v1alpha3/condition_consts.go) and use this constant both here and in the bootstrap provider